### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by thisIsTheFoxe**
Add a dark mode theme.
- Update apl-colors according to [Apple HIG](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/color/)
- Add ColorPicker to page footer
- Load svgs from file, using `@nuxtjs/svg`
  - modify svgs so that they can be adjusted using css on a container element
  - add AppStore dark svg
- Use CSS vars to dynamically change color mode
- make sure it actually looks *good*
- properly rename colors, `white: #000000` might be a bit confusing 😅

### CODE REVIEW

**Comment by thisIsTheFoxe** [comment]
Uhmmmmmm.... I have some.. questions..????
It seems to have something to do with being `disabled`. 
Cuz, surely `(110, 110, 110)` should in no way be that much **darker** than `(109, 109, 109)` and unless my eyes and color meter are broken that's not the case.. also, it seems like that's some sort of braking point.. wired things going on..
[CodePen demo](https://codepen.io/beezewax/pen/ZEKpPaW?editors=1100)

Safari screenshot of: WTF IS HAPPENING?!??!!

**Comment by moritzsternemann** [comment]
Setting the `-webkit-text-fill-color` seems to do the trick. Need to check if placeholder colors still work as expected though. See [updated CodePen](https://codepen.io/beezewax/pen/ZEKpPaW) and the [discussions on Stackoverflow](https://stackoverflow.com/questions/262158/disabled-input-text-color-on-ios) 🙂

File: components/ProfileButton.vue [hunk 1]
```diff
@@ -1,10 +1,24 @@
 <template lang="pug">
 .profile-button(:class="{ 'nuxt-link-active': linkActive }")
   .auth-pending(v-if="isAuthPending"): loading-spinner
+  //- TODO: factor svgs out somehow! make it easier to use and change coclors dynamically
```
Comment (moritzsternemann) [inline components/ProfileButton.vue]: I think we should be able to use the [`vue-svg-loader`](https://github.com/nuxt-community/svg-module#vue-svg-loader) option of `@nuxtjs/svg` to get the desired behavior.

Comment (thisIsTheFoxe) [reply components/ProfileButton.vue]: Sounds great.. but I just couldn't get it to work. Looks like once it's loaded it doesn't apply css anymore.. :/

Comment (moritzsternemann) [reply components/ProfileButton.vue]: Maybe this is caused by scoped styling and the loaded SVG not getting the correct `data-` attribute. I can try to get it working

Comment (thisIsTheFoxe) [reply components/ProfileButton.vue]: We can just change the svg's content e.g. using `currentColor` and making the sign-in-out scale to its container element.
Another thing is that we could extract them into their own component that just has the raw svg in it's template .-.

File: pages/s/_id.vue [hunk 2]
```diff
@@ -362,7 +362,7 @@ export default class ScholarProfile extends Vue {
 
     .age
       margin-left: 10px
-      color: lighten($sch-purple, 40%)
+      color: $sch-purple3
```
Comment (thisIsTheFoxe) [inline pages/s/_id.vue]: In darkmode lighten will be either almost imperceptible or basically turn white. `$sch-purple3` is a better alternative - imo.
See also, TeamCard.vue

File: components/ColorModePicker.vue [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+.color-container
+  label(v-for="color of colorSchemes", :key="color")
+    div(:class="getClasses(color.value)", @click="$colorMode.preference = color.value") {{ color.label }}
+</template>
+
+<script lang="ts">
+import { Vue } from "nuxt-property-decorator";
+
+export default class ColorModePicker extends Vue {
+  colorSchemes: { label: string; value: string }[] = [
```
Comment (thisIsTheFoxe) [inline components/ColorModePicker.vue]: not sure of the *correct* order..

File: assets/sass/imports/_colors.scss [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+//creating a function to get color instead of using css varible
+@function getColor($color-name) {
+    @return var(--color-#{$color-name});
+}
+
+//change brigthness / lightness of color *by* $amount (+ lighter, - darker)
+//$amount should be in percent 0% - 100%
+@function lightenColor($colorName, $amount) {
```
Comment (thisIsTheFoxe) [inline assets/sass/imports/_colors.scss]: not the best name I ever came up with, but I think it's understandable. Open for suggestions..

**Review by thisIsTheFoxe — COMMENTED** [review]
sorry for the `-f`s can't quite get myself to make a new commit for every minor fix that I notice.. 😅

File: assets/images/user.svg [hunk 5]
```diff
@@ -1,9 +1,7 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<svg width="32px" height="32px" viewBox="0 0 32 32" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
-    <!-- Generator: Sketch 52.6 (67491) - http://www.bohemiancoding.com/sketch -->
+<svg width="32px" height="32px" class="feather" viewBox="0 0 32 32" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (moritzsternemann) [inline assets/images/user.svg]: Why are we using fixed size for the user icon and 100% for the `sign-in-out.svg`?

File: assets/sass/imports/_colors.scss [hunk 6]
```diff
@@ -0,0 +1,169 @@
+
```
Comment (moritzsternemann) [inline assets/sass/imports/_colors.scss:2]: Would be great to have 2-space-indentation for consistency. [Editorconfig](https://editorconfig.org) is your friend 😉

File: assets/sass/imports/_vars.sass [hunk 7]
```diff
@@ -20,6 +20,9 @@
   @media (min-width: 1800px)
     @content
 
+@mixin dark-mode
+  @media (prefers-color-scheme: dark)
+    @content
```
Comment (moritzsternemann) [inline assets/sass/imports/_vars.sass]: Do we use this anywhere?

File: components/ColorModePicker.vue (hunk 3 above)
Comment (moritzsternemann) [reply components/ColorModePicker.vue]: I would say so 😁 at least it's consistent with similar view models (for example in [InputRadioGroup.vue](https://github.com/WWDCScholars/web-app/blob/main/components/inputs/InputRadioGroup.vue#L26)).

File: components/PageFooter.vue [hunk 8]
```diff
@@ -1,31 +1,40 @@
...
+    span.separator
+    a(href="https://plausible.io/wwdcscholars.com", target="_blank") Stats
+  div
+    ColorModePicker
 </template>
 
 <script lang="ts">
-import { Component, Vue } from 'nuxt-property-decorator'
+import { Component, Vue } from 'nuxt-property-decorator';
+import ColorModePicker from "./ColorModePicker.vue";
```
Comment (moritzsternemann) [inline components/PageFooter.vue]:
```suggestion
import { Component, Vue } from 'nuxt-property-decorator'
import ColorModePicker from "./ColorModePicker.vue"
```

File: components/PageFooter.vue [hunk 9]
```diff
@@ -1,31 +1,40 @@
...
 
-@Component
+@Component({
+  components: { ColorModePicker },
```
Comment (moritzsternemann) [inline components/PageFooter.vue]:
```suggestion
  components: { ColorModePicker }
```

File: components/ProfileButton.vue [hunk 10]
```diff
@@ -2,36 +2,34 @@
...
 
 <script lang="ts">
-import { Component, Vue } from 'nuxt-property-decorator'
-import { namespace } from 'vuex-class'
-import { CloudKit } from '@wwdcscholars/cloudkit'
-import LoadingSpinner from './LoadingSpinner.vue'
+import { Component, Vue } from "nuxt-property-decorator";
+import { namespace } from "vuex-class";
+import { CloudKit } from "@wwdcscholars/cloudkit";
+import LoadingSpinner from "./LoadingSpinner.vue";
```
Comment (moritzsternemann) [inline components/ProfileButton.vue]:
```suggestion
import { Component, Vue } from 'nuxt-property-decorator'
import { namespace } from 'vuex-class'
import { CloudKit } from '@wwdcscholars/cloudkit'
import LoadingSpinner from './LoadingSpinner.vue'
```

File: components/ProfileButton.vue [hunk 11]
```diff
@@ -2,36 +2,34 @@
...
 
-import * as auth from '~/store/auth'
-const Auth = namespace(auth.name)
+import User from "~/assets/images/user.svg?inline";
+import SignInOut from "~/assets/images/sign-in-out.svg?inline";
+
+import * as auth from "~/store/auth";
+const Auth = namespace(auth.name);
```
Comment (moritzsternemann) [inline components/ProfileButton.vue]:
```suggestion
import User from '~/assets/images/user.svg?inline'
import SignInOut from '~/assets/images/sign-in-out.svg?inline'

import * as auth from '~/store/auth'
const Auth = namespace(auth.name)
```

Comment (moritzsternemann) [reply components/ProfileButton.vue]: Maybe we should rename the imported SVGs to something like `UserIcon` and `SignOutIcon` 🤔

File: components/inputs/InputDate.vue [hunk 12]
```diff
@@ -1,6 +1,7 @@
 <template lang="pug">
 .input-date(:class="{ 'input-has-value': inputHasValue }")
   label
+    //- NOTE: flat-pickr does not have a darkmode!
```
Comment (moritzsternemann) [inline components/inputs/InputDate.vue]: I don't like it very much anyways.. but haven't found any good alternatives yet.

File: nuxt.config.ts [hunk 13]
```diff
@@ -131,7 +131,10 @@ const config: NuxtConfig = {
     'vue-plausible',
 
     // Load sentry
-    '@nuxtjs/sentry'
+    '@nuxtjs/sentry',
+
+    //Load color-scheme / colorMode
```
Comment (moritzsternemann) [inline nuxt.config.ts]:
```suggestion
    // Load color-scheme / colorMode
```

File: nuxt.config.ts [hunk 14]
```diff
@@ -202,9 +205,12 @@ const config: NuxtConfig = {
   buildModules: [
     // Transpile TypeScript
     '@nuxt/typescript-build',
-
+
```
Comment (moritzsternemann) [inline nuxt.config.ts]:
```suggestion

```

File: package.json [hunk 15]
```diff
@@ -49,7 +49,9 @@
   "devDependencies": {
     "@nuxt/types": "^2.15.3",
     "@nuxt/typescript-build": "^2.1.0",
+    "@nuxtjs/color-mode": "^2.0.10",
```
Comment (moritzsternemann) [inline package.json]: I think this should be a prod dependency, not 100% sure though 🙈

**Review by moritzsternemann — CHANGES_REQUESTED** [review]
I finally got around to take a thorough look. Overall dark-mode already looks amazing and I'm happy to discuss everything mentioned below!

Here are the things I noticed:
- The `apl-` colors where initially taken from the WWDC 2016 page, so don't really originate from the Apple HIG. I think since we only use the background color anyways, we should get rid of all of them and replace the background colors with our own semantic versions.
- Additionally, maybe we can improve color naming a bit.. Removing all the `sch-` prefixes and use more semantic names in general. What I came up with for the iOS app is for example `brandPrimary` and `brandSecondary` for the different shades of purple. At the same time, it would be great to get rid of all the dynamic color adjustments (`lightColor`, ...) and just have predefined colors for all these cases.
- The dark mode colors don't pop as much as their light mode counterparts. In the app, we use `rgb(165, 153, 255)` as the dark mode version of purple for example. While it might be a bit too much, I think in general the colors should be a little more vibrant in dark mode.
- I don't really like the white shadows in dark mode. Instead we should use the concept of elevation and make items on top of others (secondary nav bar, `base-section`s, profile picture borders, ...) a lighter shade of gray.
- Maybe we can improve the footer layout with the `ColorModePicker` a bit. I'm thinking of something more like developer.apple.com. _Imprint_, _Privacy Policy_, and _Stats_ links on the left, and the color mode picker on the right. What do you think?
- The rounded _Edit Profile_ button has a different fill color on hover than the border color. I think they should be the same.
- The `+` icon in the `InputImageMultiple` component is white. I can create an SVG instead, so we can color it like the user and sign-in-out icons. Same with the `upload` icon in the `InputImage` component.

File: assets/sass/imports/_vars.sass (hunk 7 above)
Comment (thisIsTheFoxe) [reply assets/sass/imports/_vars.sass]: not yet, iirc.. 👀 just thought it might be useful .-.

**Comment by thisIsTheFoxe** [comment]
hey, thanks for the review! ^^ 
Most sounds good to me.. just some other points

> Removing all the `sch-` prefixes and use more semantic names in general. What I came up with for the iOS app is for example `brandPrimary` and `brandSecondary`.

tbh, I actually kinda like the `sch-` prefixes ^^ makes it easy to distinguish brand from *generic* colors. And when working with strings I thought it might be easier to simplify "primary", "secondary", etc. with just the numbers. 

>  it would be great to get rid of all the dynamic color adjustments (lightColor, ...) and just have predefined colors for all these cases

well, I found it to be really useful from time to time to have such simple modify functions. Especially with the already dynamic `fg` / `bg` colors. Especially, again, when you get into naming `background-color-transparency-50`, `background-color-lighter-06`, etc.

> * Maybe we can improve the footer layout with the `ColorModePicker` a bit. [...] What do you think?

Initially liked it. After implementing it (at least how I did it) it doesn't really feel that good. May both in the middle with a bit of padding?

> * The rounded _Edit Profile_ button has a different fill color on hover than the border color. I think they should be the same.

Not quite sure what you mean with that.. 🤔

> * The `+` icon in the `InputImageMultiple` component is white. I can create an SVG instead, so we can color it like the user and sign-in-out icons. Same with the `upload` icon in the `InputImage` component.

I mean as you say, they're already white and work great in dark. And in light it's good as it were. Up to you.. ^^

File: components/ColorModePicker.vue (hunk 3 above)
Comment (thisIsTheFoxe) [reply components/ColorModePicker.vue]: I actually meant the order of the colors bellow lol GitHub comments are hard :D

File: assets/images/user.svg (hunk 5 above)
Comment (thisIsTheFoxe) [reply assets/images/user.svg]: Technically it's not needed. But you're right, should be 💯  for consistency

File: components/inputs/InputDate.vue (hunk 12 above)
Comment (moritzsternemann) [reply components/inputs/InputDate.vue]: removed it in #14

**Pull request merged**